### DISCUSSION TITLE
oshi dependency not able to generate system metrics when containerized

### DISCUSSION THREAD

**Discussion by mounikaVeeha** [oshi/oshi#2058] [Q&A]
We are using oshi dependency version 5.3.1 for generating system metrics .We are able to see metrics while using it with opentelemetry java auto instrumentation when we run the application locally but when we containerized it , we are not able to see the system metrics.

**Comment by dbwiddis** [comment]
Some containers need configuration to see the metrics OSHI collects.  Is this a Linux or Windows container?

**Reply by mounikaVeeha** [reply]
We are trying for windows container first .

**Reply by dbwiddis** [reply]
Which container?  See https://github.com/oshi/oshi/issues/1976 for someone with similar problems and a proposed solution

**Reply by mounikaVeeha** [reply]
We are using Kubernetes containers to deploy our applications .

**Reply by dbwiddis** [reply]
Unfortunately that's still pretty vague.  Do you have debug logs of any error messages you're getting?   What specific metrics are missing?

OSHI is designed to collect Operating System metrics.  From inside a container, some OS-level values, such as the performance counters in Windows, may not be fully accessible.  And OSHI doesn't report container-level metrics.

**Reply by mounikaVeeha** [reply]
We have an Oshi dependency in our application along with open telemetry java agent(auto instrumentation).Oshi will export system metrics like memory utilization and memory usage to otel agent which will later get exported to opentelemetry collector .This process works fine in local but not able to get the oshi metrics when the application is containerized.We need system metrics like system memory utilization and system memory usage as well accessible in containerized environment

**Reply by dbwiddis** [reply]
I expect there is probably some permissions issue associated with the container being able to read system level metrics, but without specific code details or log error messages, etc., I can't help much more.  Is your code able to be posted publicly anywhere to reproduce the problem?
